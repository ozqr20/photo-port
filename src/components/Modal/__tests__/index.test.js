// __tests__/PhotoList.test.js
import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Modal from '..';


const mockToggleModal = jest.fn();
const currentPhoto = {
    name: "Park bench",
    category: 'landscape',
    description: "lorem ipsum dolor siy amet, consectetur adispiscing elit. Bubca ultricie",
    index: 1
};

afterEach(cleanup);

describe('Modal component', () => {
    it('renders', () => {
        //Baseline render component test
        render(<Modal
            onClose={mockToggleModal}
            currentPhoto={currentPhoto}
          />);
    });
        // snapshot test
    it('Matches snapshot DOM node structure', () => {
        // Arrange the snapshot - declare variables
        const { asFragment } = render(<Modal
            onClose={mockToggleModal}
            currentPhoto={currentPhoto}
          />);
        // Assert the match 
        expect(asFragment()).toMatchSnapshot();
    });
})

describe('Click Event', () => {
    it('calls onClose handler', () => {
    // Arrange: Render Modal
      const { getByText } = render(<Modal
        onClose={mockToggleModal}
        currentPhoto={currentPhoto}
      />);
    // Act: Simulate click event
      fireEvent.click(getByText('Close this modal'))
    // Assert: Expected matcher
      expect(mockToggleModal).toHaveBeenCalledTimes(1);
    });
  })