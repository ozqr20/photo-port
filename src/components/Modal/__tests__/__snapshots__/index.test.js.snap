// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component Matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
      >
        Park bench 
      </h3>
      <img
        alt="current category"
        src="1.jpg"
      />
      <p>
        lorem ipsum dolor siy amet, consectetur adispiscing elit. Bubca ultricie
      </p>
      <button
        type="button"
      >
        Close this modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;
